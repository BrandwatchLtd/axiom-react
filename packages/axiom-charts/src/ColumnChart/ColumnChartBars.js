import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { Base } from '@brandwatch/axiom-components';

export default class ColumnChartBars extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
  };

  render() {
    const { children, ...rest } = this.props;

    return (
      <Base {...rest} className="ax-column-chart__bars">
        {children}
      </Base>
    );
  }
}
