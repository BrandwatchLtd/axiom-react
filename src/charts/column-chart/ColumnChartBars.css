@import './vars';

.ax-column-chart__bars {
  display: flex;
  flex: 1 1 0%;
  margin-right: var(--column-spacing);

  &:last-child { margin-right: 0; }
}
