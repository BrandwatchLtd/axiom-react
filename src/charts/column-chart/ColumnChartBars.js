import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { Base } from 'bw-axiom';
import './ColumnChartBars.css';

export default class ColumnChartBars extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
  };

  render() {
    const { children, ...rest } = this.props;

    return (
      <Base { ...rest } className="ax-column-chart__bars">
        { children }
      </Base>
    );
  }
}
