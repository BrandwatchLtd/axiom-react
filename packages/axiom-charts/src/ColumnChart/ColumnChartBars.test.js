import React from 'react';
import renderer from 'react-test-renderer';
import ColumnChartBars from './ColumnChartBars';

const getComponent = (props = {}) =>
  renderer.create(<ColumnChartBars {...props}>Lorem</ColumnChartBars>);

describe('ColumnChartBars', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
