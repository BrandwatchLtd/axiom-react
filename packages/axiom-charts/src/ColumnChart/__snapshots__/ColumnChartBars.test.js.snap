// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColumnChartBars renders with defaultProps 1`] = `
<div
  className="ax-column-chart__bars"
>
  Lorem
</div>
`;
